src/Publish/Microsoft.NET.Sdk.Publish.Tasks/MsDeploy/MSDeploy.cs

```diff
@@ -911,13 +911,47 @@ private static void GenerateSwitchPerItem(Utilities.CommandLineBuilder commandLi
 
 
         }
-        
+
+        internal static void IncorporateSettingsFromHostObject(ref Framework.ITaskItem[] skipRuleItems, Framework.ITaskItem[] destProviderSetting, System.Collections.Generic.IEnumerable<Framework.ITaskItem> hostObject)
+        {
+            if (hostObject != null)
+            {
+                //retrieve user credentials
+                Framework.ITaskItem credentialItem = hostObject.FirstOrDefault<Framework.ITaskItem>(p => p.ItemSpec == VSMsDeployTaskHostObject.CredentailItemSpecName);
+                if (credentialItem != null && destProviderSetting != null && destProviderSetting[0] != null)
+                {
+                    Framework.ITaskItem destSettings = destProviderSetting[0];
+                    string userName = credentialItem.GetMetadata(VSMsDeployTaskHostObject.UserMetaDataName);
+                    if (!string.IsNullOrEmpty(userName))
+                    {
+                        destSettings.SetMetadata(VSMsDeployTaskHostObject.UserMetaDataName, userName);
+                        destSettings.SetMetadata(VSMsDeployTaskHostObject.PasswordMetaDataName, credentialItem.GetMetadata(VSMsDeployTaskHostObject.PasswordMetaDataName));
+                    }
+                }
+
+                //retrieve skip rules
+                System.Collections.Generic.IEnumerable<Framework.ITaskItem> skips = from item in hostObject where (item.ItemSpec == VSMsDeployTaskHostObject.SkipFileItemSpecName) select item;
```

Comment:
Could we make this consistent with the other LINQ usages (extension method syntax)?